### PR TITLE
removing unnecessary %(get_in) template variable

### PR DESCRIPTION
**Pull request by aryamazaheri**
I created a new branch called "minor_refactoring" to inform you about some parts of the code which seem to be obsolete and unnecessary. I can find those snippets and try to fix them and push them to this branch and gradually we can merge them into the master, if you seem fit.
There might be a reason behind some of the code which I assumed to be unnecessary. In so, please let me know. Otherwise, keeping them like this will make the codebase complex and hard to follow in the future.

### CODE REVIEW

File: src/cnn_codegen.cc [hunk 1]
```diff
@@ -180,20 +180,6 @@ namespace boda
...
-	"      if(smem_in_ix_y >= 0 && smem_in_ix_x >= 0 && \n"
-	"          %%(out_pel_ix_img) < %%(in_img_dim) && \n"
-	"         smem_in_ix_x < %%(in_x_dim) && smem_in_ix_y < %%(in_y_dim) ) {\n"
-	"        v = in[%%(out_pel_ix_img)*%%(in_img_stride) +\n"
-	"          %%(filts_ix_out_chan_elem_in_chan)*%%(in_chan_stride) +\n"
-	"          smem_in_ix_y*%%(in_y_stride) +\n"
-	"          smem_in_ix_x*%%(in_x_stride)];\n" 
-	"      }"
-				       );
-      rcg->set( "get_in", get_in );
```
Comment (aryamazaheri) [inline src/cnn_codegen.cc]: I removed this part, as I understood that there is no need for such code generation. simply putting this code in the cucl code will do the job. I ran the tests and they are all passing for me.

Comment (moskewcz) [reply src/cnn_codegen.cc]: yeah, i think this is a case where it's fine either way. i guess perhaps either at some point there was some non-trivial logic on the code generation side, or perhaps it was just anticipated. anyway, sure, it's fine to inline it, although unless there's some particular purpose to do so it perhaps borders on churn. but if you want it inlined i'm fine with it; i'll merge when i can.
